instrumentation/aws-lambda-1.0/auto/src/main/java/io/opentelemetry/instrumentation/auto/awslambda/v1_0/AwsLambdaRequestHandlerInstrumentation.java

```diff
@@ -53,12 +56,28 @@
 
   @Override
   public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
-    return Collections.singletonMap(
+    // Order of transformation matters. A bit fragile, but our unit tests will catch breakage
+    // easily.
+    Map<ElementMatcher<? super MethodDescription>, String> transformers = new LinkedHashMap<>();
+
+    // First instrument the function invocation itself.
+    transformers.put(
         isMethod()
             .and(isPublic())
             .and(named("handleRequest"))
             .and(takesArgument(1, named("com.amazonaws.services.lambda.runtime.Context"))),
         AwsLambdaRequestHandlerInstrumentation.class.getName() + "$HandleRequestAdvice");
+
+    // Then instrument message handling if appropriate.
```

Comment:
If I separated out a different module, though, not sure it'd be even possible to do this pattern.

Reply:
I don't like this strict dependency between 2 instrumentations :(

Reply:
I think it's similar to our servlet instrumentation - it seems to happen when there's an overall framework, with subframeworks built on top of it. Is it that bad?

Reply:
In general inter-instrumentation dependencies are Ok, I guess. But in this case they apply to the same method and the order is important. This seems fragile

Reply:
i'm ok with current approach

the other option I see is to combine into a single advice, e.g.

```
  public static class HandleEventAdvice {
    @Advice.OnMethodEnter(suppress = Throwable.class)
    public static void onEnter(
        @Advice.Argument(value = 0, typing = Typing.DYNAMIC) Object arg,
        @Advice.Argument(1) Context context,
        @Advice.Local("otelFunctionSpan") Span functionSpan,
        @Advice.Local("otelFunctionScope") Scope functionScope,
        @Advice.Local("otelMessageSpan") Span messageSpan,
        @Advice.Local("otelMessageScope") Scope messageScope) {
      functionSpan = FUNCTION_TRACER.startSpan(context, Kind.SERVER);
      functionScope = FUNCTION_TRACER.startScope(functionSpan);
      if (arg instanceof SQSEvent) {
        messageSpan = MESSAGE_TRACER.startSpan(context, (SQSEvent) arg);
        messageScope = MESSAGE_TRACER.startScope(messageSpan);
      }
    }

    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
    public static void stopSpan(
        @Advice.Thrown Throwable throwable,
        @Advice.Local("otelFunctionSpan") Span functionSpan,
        @Advice.Local("otelFunctionScope") Scope functionScope,
        @Advice.Local("otelMessageSpan") Span messageSpan,
        @Advice.Local("otelMessageScope") Scope messageScope) {

      if (messageScope != null) {
        messageScope.close();
        if (throwable != null) {
          MESSAGE_TRACER.endExceptionally(messageSpan, throwable);
        } else {
          MESSAGE_TRACER.end(messageSpan);
        }
      }

      functionScope.close();
      if (throwable != null) {
        FUNCTION_TRACER.endExceptionally(functionSpan, throwable);
      } else {
        FUNCTION_TRACER.end(functionSpan);
      }
      OpenTelemetrySdkAccess.forceFlush(1, TimeUnit.SECONDS);
    }
  }
```

Reply:
Thanks this is better!